### PR TITLE
Upgrade to sonarlint-core 2.0-M4

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -39,7 +39,7 @@ repositories {
 
 dependencies {
   compile 'org.sonarsource.sonarlint.core:sonarlint-client-api:1.1'
-  compile 'org.sonarsource.sonarlint.core:sonarlint-core:1.1'
+  compile 'org.sonarsource.sonarlint.core:sonarlint-core:2.0-M4'
```
Comment (henryju) [inline build.gradle]: RC1 is available

File: src/main/java/org/sonarlint/intellij/issue/tracking/Trackable.java [hunk 2]
```diff
@@ -27,13 +27,11 @@
    * The line index, starting with 1. Null means that
    * issue does not relate to a line (file issue for example).
    */
-  @CheckForNull
-  Integer getLine();
+  @CheckForNull Integer getLine();
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/issue/tracking/Trackable.java]: This is not our coding style to put annotation on the same line

File: src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java [hunk 3]
```diff
@@ -19,29 +19,9 @@
...
 import com.intellij.ide.OccurenceNavigator;
 import com.intellij.ide.util.PropertiesComponent;
-import com.intellij.openapi.actionSystem.ActionGroup;
-import com.intellij.openapi.actionSystem.ActionManager;
-import com.intellij.openapi.actionSystem.ActionPlaces;
-import com.intellij.openapi.actionSystem.ActionToolbar;
-import com.intellij.openapi.actionSystem.DataProvider;
-import com.intellij.openapi.actionSystem.DefaultActionGroup;
-import com.intellij.openapi.actionSystem.IdeActions;
+import com.intellij.openapi.actionSystem.*;
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java]: We are supposed to disable import grouping

Comment (dbmeneses) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java]: Thanks, will fix. I reformatted all code using Sonar code style, but I will recheck the guidelines.

File: src/main/java/org/sonarlint/intellij/ui/nodes/FileNode.java [hunk 4]
```diff
@@ -21,10 +21,9 @@
 
 import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.vfs.VirtualFile;
-
-import javax.swing.Icon;
 import com.intellij.ui.ColoredTreeCellRenderer;
 import com.intellij.ui.SimpleTextAttributes;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/nodes/FileNode.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/ui/tree/IssueTreeCellRenderer.java [hunk 5]
```diff
@@ -20,11 +20,10 @@
 package org.sonarlint.intellij.ui.tree;
 
 import com.intellij.ui.ColoredTreeCellRenderer;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/tree/IssueTreeCellRenderer.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalConfigurable.java [hunk 6]
```diff
@@ -22,11 +22,10 @@
 import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.options.Configurable;
 import com.intellij.openapi.options.ConfigurationException;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalConfigurable.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintToolPanel.java [hunk 7]
```diff
@@ -30,12 +30,11 @@
 import com.intellij.openapi.wm.ToolWindow;
 import com.intellij.openapi.wm.ex.ToolWindowEx;
 import com.intellij.util.messages.MessageBusConnection;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintToolPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.java [hunk 8]
```diff
@@ -22,26 +22,19 @@
...
-import org.jetbrains.annotations.Nullable;
-import org.sonarlint.intellij.analysis.SonarLintFacade;
-
-import javax.swing.JComponent;
-import javax.swing.JEditorPane;
-import javax.swing.JLabel;
-import javax.swing.JPanel;
-import javax.swing.SwingConstants;
+import java.awt.*;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.java [hunk 9]
```diff
@@ -22,26 +22,19 @@
...
 import com.intellij.ui.SideBorder;
-import org.jetbrains.annotations.Nullable;
-import org.sonarlint.intellij.analysis.SonarLintFacade;
-
-import javax.swing.JComponent;
-import javax.swing.JEditorPane;
-import javax.swing.JLabel;
-import javax.swing.JPanel;
-import javax.swing.SwingConstants;
+import java.awt.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java [hunk 10]
```diff
@@ -25,18 +25,16 @@
...
-import javax.swing.table.AbstractTableModel;
-import javax.swing.table.TableCellEditor;
-import javax.swing.table.TableModel;
-import java.awt.BorderLayout;
+import java.awt.*;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.Map;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java [hunk 11]
```diff
@@ -25,18 +25,16 @@
...
 import com.intellij.util.ui.EditableModel;
-import org.jetbrains.annotations.Nullable;
-
-import javax.swing.BorderFactory;
-import javax.swing.JPanel;
-import javax.swing.table.AbstractTableModel;
-import javax.swing.table.TableCellEditor;
-import javax.swing.table.TableModel;
-import java.awt.BorderLayout;
+import java.awt.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/SonarLintColorSettingsPage.java [hunk 12]
```diff
@@ -26,16 +26,15 @@
...
-import org.jetbrains.annotations.Nullable;
-import org.sonarlint.intellij.util.ResourceLoader;
-
-import javax.swing.Icon;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Comparator;
 import java.util.Map;
 import java.util.TreeMap;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/SonarLintColorSettingsPage.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettingsPanel.java [hunk 13]
```diff
@@ -20,14 +20,10 @@
 package org.sonarlint.intellij.config.global;
 
 import com.intellij.openapi.ui.VerticalFlowLayout;
-
-import javax.swing.BorderFactory;
-import javax.swing.JCheckBox;
-import javax.swing.JComponent;
-import javax.swing.JPanel;
-import java.awt.BorderLayout;
+import java.awt.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettingsPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettingsPanel.java [hunk 14]
```diff
@@ -20,14 +20,10 @@
...
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettingsPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectConfigurable.java [hunk 15]
```diff
@@ -22,11 +22,10 @@
 import com.intellij.openapi.options.Configurable;
 import com.intellij.openapi.options.ConfigurationException;
 import com.intellij.openapi.project.Project;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectConfigurable.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java [hunk 16]
```diff
@@ -57,6 +43,20 @@
...
 import com.intellij.util.messages.MessageBusConnection;
 import com.intellij.util.ui.UIUtil;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.Collection;
+import java.util.Map;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java [hunk 17]
```diff
@@ -57,6 +43,20 @@
 import com.intellij.util.EditSourceOnEnterKeyHandler;
 import com.intellij.util.messages.MessageBusConnection;
 import com.intellij.util.ui.UIUtil;
+import java.awt.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintIssuesPanel.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: src/main/java/org/sonarlint/intellij/util/ResourceLoader.java [hunk 18]
```diff
@@ -21,14 +21,12 @@
...
-
-import javax.annotation.concurrent.ThreadSafe;
-import javax.swing.Icon;
-import javax.swing.ImageIcon;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.concurrent.ThreadSafe;
+import javax.swing.*;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/ResourceLoader.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

**Comment by SonarTech** [comment]
SonarQube analysis reported 15 issues:
* MAJOR 15 major

Watch the comments in this conversation to review them.

**Pull request merged**